libraries/exoplayer_rtsp/src/main/java/androidx/media3/exoplayer/rtsp/reader/RtpAmrReader.java

```diff
@@ -0,0 +1,193 @@
+    this.isWideBand = (payloadFormat.format.sampleMimeType == MimeTypes.AUDIO_AMR_WB);
+  }
+
+  // RtpPayloadReader implementation.
+  @Override
+  public void createTracks(ExtractorOutput extractorOutput, int trackId) {
+    trackOutput = extractorOutput.track(trackId, C.TRACK_TYPE_AUDIO);
+    trackOutput.format(payloadFormat.format);
+  }
+
+  @Override
+  public void onReceivingFirstPacket(long timestamp, int sequenceNumber) {
+    this.firstReceivedTimestamp = timestamp;
+  }
+
+  @Override
+  public void consume(
+      ParsableByteArray data, long timestamp, int sequenceNumber, boolean rtpMarker) {
+    // Check that this packet is in the sequence of the previous packet.
+    if (previousSequenceNumber != C.INDEX_UNSET) {
+      int expectedSequenceNumber = RtpPacket.getNextSequenceNumber(previousSequenceNumber);
+      if (sequenceNumber != expectedSequenceNumber) {
+        Log.w(
+            TAG,
+            Util.formatInvariant(
+                "Received RTP packet with unexpected sequence number. Expected: %d; received: %d.",
+                expectedSequenceNumber, sequenceNumber));
+      }
+    }
+    checkNotNull(trackOutput);
+    /**
+     * AMR as RTP payload RFC4867 Section-4.2
+     * +----------------+-------------------+----------------
+     * | payload header | table of contents | speech data ...
+     * +----------------+-------------------+----------------
+     *
+     * Payload header RFC4867 Section-4.4.1
+     * As interleaving is not supported currently, our header won't contain ILL and ILP
+     * +-+-+-+-+-+-+-+
+     * | CMR |R|R|R|R|
+     * +-+-+-+-+-+-+-+
+     */
+    // skip CMR and reserved bits
+    data.skipBytes(1);
+    // Loop over sampleSize to send multiple frames along with appropriate timestamp when compound
+    // payload support is added
+    int frameType = (data.peekUnsignedByte() >> 3) & 0x0f;
+    int frameSize = getFrameSize(frameType, isWideBand);
+    int sampleSize = data.bytesLeft();
+    checkArgument(sampleSize == frameSize, "compound payload not supported currently");
+    trackOutput.sampleData(data, sampleSize);
+    long sampleTimeUs =
+        toSampleTimeUs(startTimeOffsetUs, timestamp, firstReceivedTimestamp, sampleRate);
+    trackOutput.sampleMetadata(
+        sampleTimeUs,
+        C.BUFFER_FLAG_KEY_FRAME,
+        sampleSize,
+        /* offset= */ 0,
+        /* encryptionData= */ null);
+    previousSequenceNumber = sequenceNumber;
+  }
+
+  @Override
+  public void seek(long nextRtpTimestamp, long timeUs) {
+    firstReceivedTimestamp = nextRtpTimestamp;
+    startTimeOffsetUs = timeUs;
+  }
+
+  // Internal methods.
+
+  private boolean isValidFrameType(int frameType) {
+    if (frameType < 0 || frameType > 15) {
+      return false;
+    }
+    // For wide band, type 10-13 are for future use.
+    // For narrow band, type 12-14 are for future use.
+    return isWideBand ? (frameType < 10 || frameType > 13) : (frameType < 12 || frameType > 14);
```

Comment:
RFC4867 Page 19 said

>   If receiving a ToC entry with a FT value in the range 9-14 for AMR or 10-13 for AMR-WB, the whole packet SHOULD be discarded. 

I think the condition should be `(frameType < 9 || frameType > 14);`